nilearn/datasets/func.py

```diff
@@ -3053,7 +3055,7 @@ def _glob_fiac_data():
     logger.log("Data absent, downloading...")
     url = "https://nipy.org/data-packages/nipy-data-0.2.tar.gz"
 
-    archive_path = os.path.join(data_dir, os.path.basename(url))
+    archive_path = os.path.join(data_dir, Path(url).name)
```

Comment:
```suggestion
    archive_path = Path(data_dir) / Path(url).name
```

Seeing that we get `data_dir` from

```python
    data_dir = get_dataset_dir(
        "fiac_nilearn.glm", data_dir=data_dir, verbose=verbose
)
```

I think that it may good to soon (maybe in the next PR) change `nilearn.datasets._utils.get_dataset_dir` and make sure it returns pathlib.Path and adapt a lot of the code downstream from there.

Reply:
Shall I create an issue and start working on that?

Reply:
let's open a separate issue and work on it after this PR